src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -697,13 +750,20 @@ private void ParseProject(string firstLine)
             ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile(line != null, "SubCategoryForSolutionParsingErrors",
                 new BuildEventFileInfo(FullPath), "SolutionParseProjectEofError", proj.ProjectName);
 
-            // Add the project to the collection
-            AddProjectToSolution(proj);
-            // If the project is an etp project then parse the etp project file 
-            // to get the projects contained in it.
-            if (IsEtpProjectFile(proj.RelativePath))
+            if (slnFilter == null || slnFilter.Contains(proj.RelativePath))
```

Comment:
Is this valid? I would have expected that we'd need to understand all projects in the solution for configuration purposes, even if they're not valid entry points.

Reply:
When you make a solution filter file, it automatically checks for dependencies, so my understanding is that you will either have all the necessary projects in the filter file, or it should fail.